manifests/setting.pp

```diff
@@ -21,7 +21,8 @@
   validate_re($ensure,  ['file', 'present', 'absent'])
 
   unless is_integer($priority) {
-    fail('apt::setting priority must be an integer')
+    # need this to allow zero-padded priority.
+    validate_re($priority, '^\d+$', 'apt::setting priority must be an integer')
```

Comment:
We should change the error message in that case to indicate that it can be zero-padded too.